addon/components/au-icon.gts

```diff
@@ -39,16 +39,30 @@ export default class AuIcon extends Component<AuIconSignature> {
     }
   }
 
+  get iconComponent() {
+    return typeof this.args.icon !== 'string' && this.args.icon;
+  }
+
   <template>
-    <svg
-      role="img"
-      class="au-c-icon au-c-icon--{{@icon}} {{this.alignment}} {{this.size}}"
-      aria-hidden={{this.ariaHidden}}
-      ...attributes
-    >
-      <use
-        xlink:href="{{this.iconPrefix}}@appuniversum/ember-appuniversum/appuniversum-symbolset.svg#{{@icon}}"
-      ></use>
-    </svg>
+    {{#if this.iconComponent}}
+      {{#let this.iconComponent as |IconComponent|}}
+        {{! @glint-expect-error: glint seems to think that this is not a valid component type  }}
```

Comment:
I think this might be fixed by using the `ComponentLike` type.

Reply:
Hmm, it's fixed with `ComponentLike` if I don't pass any attributes.

Reply:
Yea, I think the type isn't 100% correct, and an "Element" [is needed](https://github.com/appuniversum/ember-appuniversum/pull/481#discussion_r1524600624).